apps/aecore/src/aec_blocks.erl

```diff
@@ -292,32 +293,43 @@ hash_internal_representation(B = #block{}) ->
     aec_headers:hash_header(to_header(B)).
 
 
-%% TODO: implement validation of microblocks
--spec validate(block(), binary()) -> ok | {error, term()}.
-validate(Block, _LeaderKey) ->
+validate_key_block(Block) ->
+    case aec_headers:validate_key_block_header(to_header(Block)) of
+        ok ->
+            Validators = [fun validate_no_txs/1,
+                          fun validate_empty_txs_hash/1],
+            case aeu_validation:run(Validators, [Block]) of
+                ok              -> ok;
+                {error, Reason} -> {error, {block, Reason}}
+            end;
+        {error, Reason} ->
+            {error, {header, Reason}}
+    end.
+
+validate_no_txs(#block{txs = Txs}) ->
+    case Txs =:= [] of
+        true  -> ok;
+        false -> {error, txs_in_key_block}
+    end.
+
+validate_empty_txs_hash(#block{txs_hash = TxsHash}) ->
+    case TxsHash =:= <<0:?TXS_HASH_BYTES/unit:8>> of
```

Comment:
A would prefer having a callback in aec_txs_trees to check if this is the empty hash.

Reply:
Nevermind, this will go away later.